Enable emulated hue bridge to use sleep cycle

orrpan opened the pull request:
## Description: Enable emulated hue bridge to use sleep cycle, new try #14825
Based on this [https://github.com/magicus/ha-local-echo](https://github.com/magicus/ha-local-echo) and on this discussion [https://community.home-assistant.io/t/getting-iphone-apps-that-support-hue-to-work-with-hue-emulator/4409](https://community.home-assistant.io/t/getting-iphone-apps-that-support-hue-to-work-with-hue-emulator/4409)

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#8041

## Example entry for `configuration.yaml` (if applicable):
```yaml
emulated_hue:
  listen_port: 80
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

No new dependencies added. 

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

balloob committed:
Add Hass.io user headers to supervisor proxy

balloob committed:
Update test_http.py

orrpan committed:
Enable emulated hue bridge to use sleepcycle

orrpan committed:
Fix wrong indentation

orrpan commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -112,6 +121,11 @@ def get(self, request, username):
+        # Different response if extra_urls is used, example for sleep cycle
+        if str(request.path) != self.url.replace('{username}', username):
+            json_response = \
+            {'lights': json_response, 'config': {'mac': '00:00:00:00:00:00'}}
```
d3db83a42dbec29d8309864b38c198029de8d193

orrpan commented:
Test possible with this configuration, if not having a light with the possibility to change brightness.
```yaml
emulated_hue:
  listen_port: 80
  expose_by_default: false
  exposed_domains:
    - light
  entities: 
    light.test_light:
      hidden: false

input_number:
  test_brightness:
    name: 'Test Brightness'
    initial: 0
    min: 0
    max: 255
    step: 1

light: 
  - platform: template
    lights:          
      test_light:
        friendly_name: "test_light"
        value_template: "{{ states('input_number.test_brightness') | int >= 1 }}"
        level_template: "{{ states('input_number.test_brightness') | int }}"
        turn_on:
          service: input_number.set_value
          data:
            entity_id: input_number.test_brightness
            value: 255
        turn_off:
          service: input_number.set_value
          data:
            entity_id: input_number.test_brightness  
            value: 1
        set_level:
          - service: input_number.set_value
            data_template:
              entity_id: input_number.test_brightness
              value: "{{ brightness }}"
```

dmulcahey committed:
check config instead of config_entry for quirks flag (#19730)

syssi committed:
Don't slugify unique id (#19770)

Adminiuga committed:
Use manufacturer id only for configure_reporting only when specified. (#19729)

OttoWinter committed:
Fix some ESPHome race conditions (#19772)

* Fix some ESPHome race conditions

* Remove debug

* Update requirements_all.txt

* :ambulance: Fix IDE line length settings

cdce8p committed:
Update HAP-python to 2.4.2 (#19776)

* Bugfixes for connection issues

balloob commented:
What is Sleep Cycle and why does it not support the normal Hue API? The changes here make no sense, a new type is added to the config but it's not used, a url with `(null)` is added but that's invalid, and no tests are added.

orrpan committed:
removed unnecessary extra_urls

orrpan commented:
@balloob 
Sleep Cycle is an alarm app for phones. It uses the Hue for wake up. As in the link in the description it does not work with existing component. 

I thought the type was going to be removed, so I made it work without it. But I can change it to use the type to make correct json_response.

For the url with (null), it was used as that in the github link in the description. But now I tested without it and it seems to work without it. So my bad.

Thought it was such small change but I'll add tests.

The things Sleep Cycle needs are, a get function in `HueUsernameView`. Extra_url in `HueAllLightsStateView` with a special `json_response`, which include a dummy mac. It also needs the brightness for lights to be possible to set to 0, since it turns it off and then sets brightness to 0

OttoWinter committed:
Add ESPHome native API discovery (#19399)

* ESPHome discovery

* Add note about netdisco

* :abcd:

* Address comments

* Bump netdisco to 2.3.0

* Update requirements_all.txt

eliseomartelli committed:
Rename air pollutants to air quality (#19448)

* mv component folder

* moved in airquality

* changed names in files

* renamed test init

* renamed test air quality

* renamed in tests

* renamed coverage

* fixed naming

* corrected attr names

* changed attr names

cdheiser committed:
Fix a bug in Lutron RadioRA2 Scene support (#19819)

OttoWinter committed:
Bump aioesphomeapi (#19838)

fredrike committed:
fixes #19814, Daikin config setting (#19823)

syssi committed:
Simplify data_key for a stable unique_id because the order of the dict will not be preserved (Closes: #13522) (#19766)

orrpan committed:
move json_response wrapper to __init__.py, type now needed

orrpan commented:
> For the url with (null), it was used as that in the github link in the description. But now I tested without it and it seems to work without it. So my bad.

This was wrong, tried to reinstall the app and add the emulated hue again and it did not work. So they are necessary for Sleep Cycle at initial setup at least.

orrpan committed:
re-added extra_urls to HueUsernameView, as it does not work without it

orrpan committed:
Correcting formatting and added tests

orrpan committed:
add empty line

emontnemery committed:
Cleanup if discovered mqtt sensor can't be added (#19745)

* Cleanup if discovered mqtt sensor can't be added

* No bare except

* Clear ALREADY_DISCOVERED list with helper

emontnemery committed:
Merge pull request #19744 from emontnemery/mqtt_discovery_cleanup_binary_sensor

Cleanup if discovered mqtt binary_sensor can't be added

emontnemery committed:
Cleanup if discovered mqtt cover can't be added (#19743)

* Cleanup if discovered mqtt cover can't be added

emontnemery committed:
Cleanup if discovered mqtt alarm can't be added (#19742)

* Cleanup if discovered mqtt alarm can't be added

emontnemery committed:
Cleanup if discovered mqtt fan can't be added (#19741)

* Cleanup if discovered mqtt fan can't be added

emontnemery committed:
Cleanup if discovered mqtt climate can't be added (#19739)

* Cleanup if discovered mqtt climate can't be added

dmulcahey committed:
Relay events for onoff and levelcontrol output clusters in ZHA (#19863)

* auto relay events for onoff and levelcontrol output clusters

* fix docstring

* correct copy/paste failure - review comment

* add space - review comment

larsvinc committed:
Corrected formating.

larsvinc committed:
Added missing operator.

StevenLooman committed:
Fix error when trying to log used UPnP device, if multiple found (#19875)

StevenLooman committed:
Fix error when trying to log used UPnP device, if multiple found (#19875)

BaQs committed:
Replace influxdb query by another query that is more lightweight (#19880)

same as #6289

thomasloven committed:
Allow other icon prefixes than mdi: (#19872)

exxamalte committed:
Geo Location -> Geolocation (class names and unit test comments) (#19877)

* fixed geolocation naming in class names

* fixed geolocation naming in comments in unit test

alistairg committed:
Resolves #17196, Resolves #18739 - Hue Beyond light fixture errors (#19874)

* Resolves #17196, Resolves #18739 - Hue Beyond light fixtures being incorrectly recognized

* Removed long code lines that were failing code review

* Removed trailing whitespace

alistairg committed:
Resolves #17196, Resolves #18739 - Hue Beyond light fixture errors (#19874)

* Resolves #17196, Resolves #18739 - Hue Beyond light fixtures being incorrectly recognized

* Removed long code lines that were failing code review

* Removed trailing whitespace

pvizeli committed:
Update OZW to 0.1.2 (#19878)

* Update ozw 0.1.2

* Update requirements_all.txt

pvizeli committed:
Update OZW to 0.1.2 (#19878)

* Update ozw 0.1.2

* Update requirements_all.txt

Rendili committed:
add entity support to hive (#19879)

balloob committed:
Bumped version to 0.85.0b1

FlorianLudwig committed:
assign user to websocket connection when using legacy_api_password (#19797)

FlorianLudwig committed:
assign user to websocket connection when using legacy_api_password (#19797)

rohankapoorcom committed:
Migrate geofency over to the Webhook component (#18951)

* Migrate geofency over to the Webhook component

* Return web.Response correctly

* Fix test

* Lint

* Fix error that tests caught

balloob committed:
Refactor motion sensor of the xiaomi_aqara platform (#19805)

* Refactor motion sensor

* Improve debug output

balloob committed:
Fix deprecation warning (#19882)

Rendili committed:
Bug fix with getting a device battery level when API unavailable for Hive (#19841)

* hive updates - bug fix and add entity registration

* remove hive entity registration code

balloob committed:
Updated frontend to 20190109.0

balloob committed:
Updated frontend to 20190109.0

balloob committed:
Bumped version to 0.85.0

balloob committed:
Update translations

balloob committed:
Merge pull request #19897 from home-assistant/rc

0.85.0

bachya committed:
Bump pyflunearyou to 1.0.1 (#19899)

tokarev-artem committed:
Removed mkdir, If the WORKDIR doesn’t exist, it will be created. (#19892)

RomRider committed:
min_max sensor support for STATE_UNAVAILABLE

royduin committed:
Plant monitor defaults (#19891)

* Plant monitor defaults

* houndci-bot fixes

mitchellrj committed:
Correctly map Nest hvac_state to Home Assistant states. (#19895)

Adminiuga committed:
Don't map LevelControl to light for single cluster devices. (#19929)

Adminiuga committed:
Don't map LevelControl to light for single cluster devices. (#19929)

cliffordwhansen committed:
Add btle_name attribute to devices (#19915)

* Update googlehome.py

Added name from bluetooth device to attributes

* Update homeassistant/components/device_tracker/googlehome.py

Check if key exists before assigning name

Co-Authored-By: cliffordwhansen <clifford@nighthawk.co.za>

fabaff committed:
Upgrade youtube_dl to 2019.01.10

dshokouhi committed:
Fix botvac connected alert retrieval

dshokouhi committed:
Fix botvac connected alert retrieval

abmantis committed:
Add Alexa's EndpointHealth reporting (#19784)

* add Health reports

* add health report for all devices

* update tests

* Update homeassistant/components/alexa/smart_home.py

Co-Authored-By: abmantis <abmantis@users.noreply.github.com>

* lint

* add tests

nickneos committed:
Remove Discovery dependency from konnected.py (#19910)

arigilder committed:
Add sensors to jewish_calendar for upcoming Shabbat times (#19278)

* Initial pass of cleanup for shabbat_times

* Switch to async defs

* First pass of unit tests + fixture data

* Completion of first round of unit tests, 100% passing

* Unit tests for state restoring

* Style fixes

* More style fixes

* Lint fix

* Add upcoming candelighting and havdalah sensors

* Add unit tests, remove havdalah offset

* More unit tests + small bugfix for weekly_portion

* Add issur melacha sensor

* Remove old shabbat_times work-in-progress files

* Bump required version of hdate

* Add havdalah offset config parameter

* Bump hdate version required

* Pin hdate requirement

* Lint fixes

* Changes based on review + API changes for hdate 0.8.7

* Add three-day holiday unit tests

* Remove debugging line

* Add missing docstring

* Fix doc lint comment

SNoof85 committed:
add_entities -> async_add_entities (#19943)

Better state handling at HA startup

rymsha committed:
Add nad telnet (#19704)

* fix cla-bot

* fix bug introduced after linter complaint

* merge two components into one
support telnet port configuration

* remove obsolete nadtcp component. nad component must be used instead.

* back to correct nad_receiver version

fronzbot committed:
Upgrade blinkpy and use calibrated temperature for sensor (#19723)

balloob commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -112,7 +121,7 @@ def get(self, request, username):
                     self.config, entity, state, brightness)
 
-        return self.json(json_response)
+        return self.json(self.config.response_wrapper(json_response))
```
This makes no sense. Why would the sleep app implement a different Hue API than the others ? 

Here is the documentation, and this is what we need to follow: https://developers.meethue.com/develop/hue-api/7-configuration-api/#get-full-state

balloob replied:
I see it now, you are trying to merge the responses of 2 endpoints in 1 class, please split into a new endpoint.

balloob commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -88,6 +96,7 @@ class HueAllLightsStateView(HomeAssistantView):
 
     url = '/api/{username}/lights'
     name = 'emulated_hue:lights:state'
+    extra_urls = ['/api/{username}']
```
These two endpoints have different responses. You can't merge them.

abmantis committed:
Small refactoring for the alexa component (#19782)

* small refactoring

* fix tests

fabaff committed:
Upgrade pylast to 3.0.0 (#19938)

balloob committed:
Fix botvac connected alert retrieval (#19937)

therve committed:
Remove spurious libzwave error (#19928)

If a network_key is not configuired, the following error is logged:
TypeError: expected bytes, NoneType found
Exception ignored in: 'libopenzwave.str_to_cppstr'
TypeError: expected bytes, NoneType found

We don't need to set the key if it's None, let's skip in that case.

Commit:
Add emulated_roku component (#17596)

* Add emulated_roku component

* Add emulated_roku config tests

* Fix emulated_roku test dependencies

* Remove emulated_roku yaml support, add tests

* Add yaml support, simplify config flow

* Improve emulated_roku code quality

* Fix emulated_roku translation, improve code quality

* Fix emulated_roku translation

* Bump emulated_roku to 0.1.6 to fix SSDP discovery

* Bump emulated roku to 0.1.7, refactor component start/stop methods

Commit:
Change state() to try/except to catch KeyError (#19935)

* Change state() to try/except to catch KeyError

When Tautulli is up but Plex is down, the API doesn't return a 'stream_count' key. This causes calls to state() to raise KeyError exceptions. The new code includes a try/except to catch the KeyError and return -1 signifying that the Tautulli API cannot talk to Plex

* Update tautulli.py

awarecan committed:
Only authenticate request when owner can be found

fabaff committed:
Upgrade mutagen to 1.42.0 (#19956)

fabaff committed:
Upgrade pillow to 5.4.1 (#19958)

fabaff committed:
Upgrade ruamel.yaml to 0.15.85 (#19960)

fabaff committed:
Upgrade pysnmp to 4.4.8 (#19961)

fabaff committed:
Upgrade keyring to 17.1.1 (#19962)

soldag committed:
Add support for restoring state to rpi_gpio_pwm (#19944)

balloob committed:
min_max sensor support for STATE_UNAVAILABLE (#19914)

eliseomartelli committed:
add service type in name (#19980)

balloob committed:
Add support for spot cleaning that was introduced in pybotvac 0.12 (#19857)

* Added support for spot cleaning that was introduced in pybotvac 0.12.

* Corrected formating.

* Added missing operator.

balloob committed:
Merge pull request #19959 from home-assistant/awarecan-trusted-network-auth-must-have-user

Trusted Network Auth: only authenticate request when owner can be found

balloob committed:
Fix tests

syssi committed:
Support next generation of the Xiaomi Mi Smart Plug (chuangmi.plug.m3) (#19972)

* Support next generation of the Xiaomi Mi Smart Plug (chuangmi.plug.m3)

* Fix indent

jarlehjortland committed:
When tradfri experience communication errors make the lights/devices unavailable. (#19288)

balloob committed:
Allow embedded platforms (#19948)

* Allow embedded platforms

* Fix test

dmulcahey committed:
Add services and helper functions to support a config panel for ZHA (#19664)

* reconfigure zha device service

add log line to reconfigure service for consistency

* add entity functions to support new services

* added new services and web socket api and split them into their own module

* support manufacturer code

logging to debug

get safe value for manufacturer

* update services.yaml

* add comma back

* update coveragerc

* remove blank line

* fix type

* api cleanup - review comments

* move static method to helpers - review comment

* convert reconfigure service to websocket command - review comment

* change path

* fix attribute

balloob committed:
Updated frontend to 20190109.1

balloob committed:
Updated frontend to 20190109.1

balloob committed:
Fix warning (#19946)

* Fix warning

* Update service.py

balloob committed:
Fix warning (#19946)

* Fix warning

* Update service.py

balloob committed:
Fix warning (#19946)

* Fix warning

* Update service.py

balloob committed:
Bumped version to 0.85.1

balloob committed:
Merge pull request #19986 from home-assistant/rc

0.85.1

grea09 committed:
Adding IPv6 to fail2ban sensor (#19457)

* Fixing fail2ban regex for ipv6

* Adding IPv6 tests for fail2ban

* Formating code for hound

* Formating again

* Formating again 2

balloob committed:
Update test_auth.py

mikeage committed:
Always use datetime and timedelta in camera.proxy instead of int/float (#19571)

Commit:
Include Scripts/ directory to .gitignore - this is created by virtualenv on Windows (#18918)

MartinHjelmare committed:
Add mysensors state update delay (#18891)

* Add mysensors state update delay

* Schedule state update after delay to avoid updating state multiple
  times during the same short timespan.

* Code review

oblogic7 committed:
Add ability to monitor relay events (#18730)

* Add ability to monitor relay events

* Account for empty events array instead of none.

fredrike committed:
Add support for 'via_hub' for device_info (#19454)

* Add support for 'via_hub'

* Update config schema

* add domain to via_hub

* add tests for via_hub

achatham committed:
Fix the anthemav component by removing a debugging line. (#19979)

This lane ended up calling vars(transport) on an asyncio Transport
object. In a standard setup, that's a python object provided by syncio,
and it works. Home Assistant injects uvloop into asyncio, which makes this
a Python C object, and those don't support vars().

basbl committed:
Expose more information about shipments by PostNL (#18334)

* Expose more information about shipments by PostNL

* Update postnl.py

balloob committed:
Fix fail2ban tests

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Bumped version to 0.86.0.dev0

rohankapoorcom committed:
Split locative to a separate component (#19964)

* Split locative to a separate component

* Switch tests to use constants for http codes

* Fix tests

pbalogh77 committed:
Support for multiple Fibaro gateways (#19705)

* Preparing for transition to config flow

Added multiple gateway support
Reworked parameter flow to platforms to enable multiple controllers
Breaking change to config, now a list of gateways is expected instead of a single config

* Updated coveragerc

Added new location of fibaro component

* Fixes based on code review and extended logging

Addressed issues raised by code review
Added extended debug logging to get better reports from users if the device type mapping is not perfect

* Changhes based on code review

Changes to how configuration is read and schemas
Fix to device type mapping logic

* simplified reading config

* oops

oops

* grr

grr

* change based on code review

* changes based on code review

changes based on code review

balloob committed:
Add Hass.io user headers to supervisor proxy (#19395)

* Add Hass.io user headers to supervisor proxy

* Update test_http.py

* Fix tests

* Update test_auth.py

scop committed:
Upgrade huawei-lte-api to 1.1.3 (#19987)

scop committed:
Upgrade pytest-cov to 2.6.1 (#19988)

ledor473 committed:
Wink: Update pubnubsub-handler version to make it compatible with python 3.7 (#19625)

balloob committed:
Lint

Adminiuga committed:
Don't set friendly_name in Zha entity. (#19991)

Use @property name instead of setting friendly_name device state attr.

adrum committed:
Add support for HomeKit Controller Locks (#19867)

* Added HomeKit Controller Lock

* cleaned up code according to standards

* fixed lint issues

* added private constant for jammed state

* removed state_unknown

dmulcahey committed:
Repackage ZHA component (#19989)

* move files

* rename files

* rename files

* move files

* relative import

* update coveragerc

vaidyasr committed:
'latest_dir' referenced before assignment (#19952)

* 'latest_dir' referenced before assignment

local variable 'latest_dir' referenced before assignment

* Better fix

thomasdelaet committed:
catch TypeError's in addition to ValueError's for unifi direct device tracker (#19994)

* catch TypeError's in addition to ValueError's in response from unifi access point

sometimes unifi's access point returns incomplete json which results in a TypeError because ssid_table is None

* fix syntax error

quazzie committed:
Support for html5 notifications to suggest their names (#19965)

* support for devices to suggest their names

* houndci fixes

* Lint

oblogic7 committed:
Update doorbird events to include URLs on event_data (#19262)

* Update doorbird events to include URLs on event_data as shown in documentation.

(cherry picked from commit 2405bc96fe1b3a93ea8466e48db6b7df8df02235)

* Format timestamp

* Update timestamp

* Lint

jkeljo committed:
Upgrade greeneye_monitor to 1.0 (#19631)

* Upgrade greeneye_monitor to 1.0

This is a breaking change; it causes the `serial_number` field in
configuration to be treated as the full 8-digit serial number rather
than the last 5 digits as was previously done, which results in the unique
identifiers for the sensors being different. (Fixing them up in
`config/.storage/core.entity_registry` before rebooting into the updated
version seems to prevent any weirdness.)

The last-5-digits behavior was a result of me misunderstanding the packet
format docs and not realizing that the true serial number was split across
two fields. In addition to being confusing (see
https://community.home-assistant.io/t/brultech-greeneye-issues/86852), it
was technically incorrect. The `greeneye_monitor` platform was just introduced
in 0.82, so it seems like the kind of thing that's best to fix now while
adoption is relatively low rather than later when somebody runs into it
as more than just a point of confusion.

* Switch to 8-character string

* Coerce to int

* Remove now-unnecessary cast

dmulcahey committed:
version bump for zha-quirks (#20019)

carstenschroeder committed:
Enable bool type for ADS service (#20011)

Kane610 committed:
UniFi - Fix issue with POE switch reset switch config (#20021)

* Fix issue when controlling POE would reset configuration for all other ports on same device

ehendrix23 committed:
Add service change_channel to Harmony component (#19649)

* Update requirements

Updated requirements

* Add attributes

Add firmware and config version attributes

* Small bump for aioharmony

Small version bump increase for aioharmony

* Order ATTR constants

* Add the service

Add service change_channel

* Fix requirements file

For some reason aioharmony ended up in there as a duplicate. Fixed it.

* Updates based on review

sdague committed:
mychevy: Fix wrong attribute on battery level selector (#20016)

The battery level sensor is broken because the logic for determining
if the battery is charged accessed the wrong variable. This one
character fix makes it work again.

ehendrix23 committed:
Add harmony service to remote services.yaml (#20031)

Added the harmony_change_channel service to services.yaml for remote component.

rohankapoorcom committed:
Switch locative to use the webhook component

rohankapoorcom committed:
Lint

rohankapoorcom committed:
Remove dead test code

finish06 committed:
update to pyunifi 2.16 (#20042)

* update to pyunifi 2.16

* update requirements to version 2.16

scop committed:
Upgrade pytest to 4.1.0 (#20013)

arsaboo committed:
Change alarm panel code format (#20037)

* Change code format

* Update elkm1 code format

* Update alarmdecodes code_format

* Update alarmdotcom code_format

* Update concord232 code_format

* Update envisalink code_format

* Update ialarm code_format

* Update ifttt code_format

* Update manual alarm code_format

* Update manual mqtt code_format

* Update mqtt code_format

* Update ness code_format

* Update nx584 code_format

* Update satel_integra code_format

* Update simplisafe code_format

* Update verisure code_format

* Change text to be consistent with the Polymer PR

carstenschroeder committed:
ADS service: Enable use of templates for value (#20024)

OttoWinter committed:
Fix errors in ESPHome integration (#20048)

* Fix Home Assistant State Import

* Fix cover state

* Fix fan supported features

* Fix typo

amelchio committed:
Fix remote.harmony_change_channel services.yaml indentation (#20051)

balloob committed:
Update translations

balloob committed:
Updated frontend to 20190113.0

oblogic7 committed:
Show persistent notification on Doorbird schedule failure (#20033)

* Remove unnecessary return.
Add persistent notification on failure to configure doorbird schedule.

* Update doorbirdpy to 2.0.5

* Fix bare except

* Bump version again

* Lint

* Return false

adrum committed:
Add support for HomeKit Controller covers (#19866)

* Added support for HomeKit Controller covers

* removed copied code

* more linting fixes

* added device type to service info

* added checks for value in characteristics

* added state stopped parsing

* removed logger

* removed unused args

* fixed inits

* removed unused imports

* fixed lint issues

* fixed lint issues

* remove state_unknown

* remove validation of kwargs in homekit controller covers

* guarantee tilt position is not none before setting

rohankapoorcom committed:
Use voluptuous to validate the webhook schema

rohankapoorcom committed:
Validate test mode schema as well

rytilahti committed:
warning -> debug, this should not have been visible to users (#20061)

rohankapoorcom committed:
Lint

rohankapoorcom committed:
Remove allow_extra

rohankapoorcom committed:
Split out gpslogger into a separate component and platform (#20044)

* Split out gpslogger into a separate component and platform

* Lint

* Lint

* Increase test coverage

rohankapoorcom committed:
Return web.Response correctly

soberstadt committed:
Add Roku hub and remote (#17548)

* add roku remote component

* remove name config (for now)

* update coveragerc and requirements_all

* fix linting errors

* remove extra requirements entry

* fix flake8 errors

* remove some references to apple tv

* remove redundant REQUIREMENTS

* Update requirements_all.txt

* Pass hass_config to load_platform

* don't expose registry constant

* remove unnecessary registry list

* use await instead of add_job

* use ensure_list

* fix code style

* some review fixes

* code style fixes

* stop using async

* use add with update

* fix whitespace

* remove I/O from init loop

* move import

scop committed:
Upgrade pytest to 4.1.1 (#20088)

balloob committed:
Lowercase code format (#20077)

ttroy50 committed:
fix logic error in dubln bus (#20075)

shred86 committed:
Bump abode to 0.15.0 (#20064)

Fix for motion sensor states

arsaboo committed:
Change return text code for alarm control panels (#20055)

* Change return text code for alarmdotcom

* Change return text code for ialarm

* Change return text code for IFTTT

* Change return text code for manual alarm panel

* Change return text code for manual MQTT alarm

* Change return text code for MQTT

* Change return text code for Simplisafe

bachya committed:
Adjust OpenUV integration for upcoming API limit changes (#19949)

* Adjust OpenUV integration for upcoming API limit changes

* Added fix for "Invalid API Key"

* Bugfix

* Add initial nighttime check

* Move from polling to a service-based model

* Fixed test

* Removed unnecessary scan interval

* Fixed test

* Moving test imports

* Member comments

* Hound

* Removed unused import

mopolus committed:
Fix ihc issues caused by update to defusedxml (#20091)

* Update __init__.py

* Update __init__.py

OttoWinter committed:
Move ESPHome Source Files (#20092)

* Move ESPHome source files

* Update .coveragerc

* Update CODEOWNERS

bachya committed:
Embed RainMachine platforms into the component (#20066)

* Embed RainMachine platforms into the component

* Updated CODEOWNERS

* Updated .coveragerc

bachya committed:
Embed SimpliSafe platforms into the component (#20069)

* Embed SimpliSafe platforms into the component

* Updated CODEOWNERS

* Updated .coveragerc

bachya committed:
Embed OpenUV platforms into the component (#20072)

* Embed OpenUV platforms into the component

* Updated CODEOWNERS

* Updated .coveragerc

emontnemery committed:
Reconfigure MQTT lock component if discovery info is changed (#19468)

* Reconfigure MQTT lock component if discovery info is changed

* Use dict[key] for required config keys and keys with default config schema values.

andrewsayre committed:
Added partial detection to async_add_job (#20119)

fredrike committed:
Fix tellduslive discovery and auth issues (#20023)

* fix for #19954, discovered tellsticks shows up to be configured

* fix for #19954, authentication issues

* updated tests

* move I/O to executer thread pool

* Apply suggestions from code review

Co-Authored-By: fredrike <fredrik.e@gmail.com>

rohankapoorcom committed:
Use voluptuous to perform validation for the geofency webhook (#20067)

* Use voluptuous to perform validation for the geofency webhook

* Add missing attribute to schema

* Lint

eliseomartelli committed:
fixed gtt to report isotime (#20128)

emontnemery committed:
Move MQTT platforms under the component (#20050)

* Move MQTT platforms under the component

emlove committed:
Split time_pattern triggers from time trigger (#19825)

* Split interval triggers from time trigger

* Default smaller interval units to zero

* Rename interval to schedule

* Rename schedule to time_pattern

Kane610 committed:
Change deCONZ to embedded platforms (#20113)

Move all platforms into components/deconz

starkillerOG committed:
Improve Philips Hue color conversion 2 (#20118)

* Add gamut capability to color util

* Include gamut in hue_test

* Improve Philips Hue color conversion

* correct import for new location hue.light

* include file changes between PR's

* update aiohue version

* update aiohue version

* update aiohue version

* fix hue_test

Now Idea why it failed compared to the previous time

* Include gamut in hue_test

* fix hue_test

* Try to test hue gamut conversion

supply a color that is well outside the color gamut of the light, and see if the response is correctly converted to within the reach of the light.

* switch from gamut A to gamut B for the tests.

* remove white space in blanck line

* Fix gamut hue test

* Add Gamut tests for the util.color

* fix hue gamut test

* fix hue gamut test

* Improve Philips Hue color conversion

keslerm committed:
Enable setting alarm mode night for arlo platform

emontnemery committed:
Minor refactoring of MQTT availability (#20136)

* Small refactor of MQTT availability

* Use dict[key] for required config keys and keys with default values.

keslerm committed:
Missed one small comment

quazzie committed:
Add notify.html5_dismiss service (#19912)

* Add notify.html5_dismiss service

* fix test

* add can_dismiss

* fix service data payload

* fix hasattr -> getattr

* fixes

* move dismiss service to html5

* fix services.yaml

* fix line to long

balloob committed:
Updated frontend to 20190115.0

balloob committed:
Fix loading translations for embedded platforms (#20122)

* Fix loading translations for embedded platforms

* Update doc string

* Lint

rohankapoorcom committed:
Switch geofency tests to using an unauthenticated HTTP client (#20080)

Adminiuga committed:
Zha light.turn_on service fixes. (#20085)

Set color only if light supports color mode.
Set color temp only light supports color temp.
Update entity's brightness only if Zigbee command to set the brightness
was sent successfuly.

fabaff committed:
Switch to ipapi.co (fixes #19846) (#19886)

* Switch to ipapi.co (fixes #19846)

* Fix name

* Update name

amelchio committed:
Accept both domains and entities in influxdb include (#19927)

* Accept both domains and entities in influxdb include

* Explicit test

* Remove lint

rohankapoorcom committed:
#20043: Remove superfluous dict in WEBHOOK_SCHEMA validation

Kane610 committed:
Make all deCONZ platforms use a common base (#20137)

* Make all deconz platforms use a common base

* Fix one cover to device replacement too much...

ehendrix23 committed:
Reconnect and device name fix for harmony platform (#20108)

* Update requirements

Updated requirements

* Add attributes

Add firmware and config version attributes

* Small bump for aioharmony

Small version bump increase for aioharmony

* Fix requirements file

For some reason aioharmony ended up in there as a duplicate. Fixed it.

* Fix for send command with named device

* Update requirements to get reconnect fix

* Set aioharmony version to 0.1.4

Version 0.1.4 has additional small fixes.

* Keep trying to connect on startup

Keep trying to connect to the HUB on startup

* Revert rebase changes

Revert some changes that should have been reverted back as part of rebase.

* PlatformNotReady if unable to connect on startup

Will call PlatformNotReady if unable to connect to Hub on startup

* Increase aioharmony requirement to 0.1.5

Increase aioharmony requirement to 0.1.5

* Register callbacks when entity added to HASS

Register the callbacks only once the entity has been added to HASS instead of during setup of platform.

* Removed debug log in __init__

Removed debug log in __init__

balloob committed:
Fix TTS say config validation (#20145)

tgerla committed:
Use the correct Unicode degree symbol (#20058)

The previous symbol used for degrees was U+00BA, the "Masculine Ordinal Indicator". This patch changes the symbol to U+00B0, "Degree Sign", to match the rest of the Home Assistant system.

rohankapoorcom committed:
Add support for connecting to multiple zoneminder instances (#19955)

* Add support for connecting to multiple zoneminder instances

* Lint

* Lint

* Clean up and better error handling

* Fix config validation

* Lint

rohankapoorcom committed:
Embed geofency platform into component (#20083)

* Embded geofency platform into component

* #20083: Change docstring for geofency device tracker platform

pvizeli committed:
Remove .isort because we use the config from setup.cfg (#20158)

cvwillegen committed:
Fix link to documentation link, select read-only (#20155)

- Fixed link to documentation page at the top of the sensor
- Select the IMAP4 Inbox read-only so that reading e-mail does not mark messages as Read

rohankapoorcom committed:
Embed mailgun platform into component (#20147)

* Embed mailgun platform into component

* #20147: Update .coveragerc

* #20147 update requirements.txt

balloob committed:
Enable setting alarm mode night for arlo platform (#20143)

* Enable setting alarm mode night for arlo platform

* Missed one small comment

Anonym-tsk committed:
Support device_class for rest sensor (#20132)

* Ready

* Tests fixed

rohankapoorcom committed:
Migrate gpslogger to the automatically generated webhook (#20079)

* Migrate gpslogger to the automatically generated webhook

* Lint

* Lint and return error code

cvwillegen committed:
imap_email_content: allow configuring folder to read. (#20160)

* Update imap_email_content.py

Add configuration for selection of IMAP folder to track

* Update imap_email_content.py

dmulcahey committed:
Make imports relative in ZHA component (#20020)

* make imports relative

* remove cyclic import

balloob committed:
Switch locative to use the webhook component (#20043)

* Switch locative to use the webhook component

* Lint

* Remove dead test code

* Use voluptuous to validate the webhook schema

* Validate test mode schema as well

* Lint

* Remove allow_extra

* Return web.Response correctly

* #20043: Remove superfluous dict in WEBHOOK_SCHEMA validation

emontnemery committed:
Log exceptions thrown by MQTT message callbacks (#19977)

* Log exceptions thrown by MQTT message callbacks

* Fix tests

* Correct method for skipping wrapper in traceback

* Lint

* Simplify traceback print

* Add test

* Move wrapper to common helper function

* Typing

* Lint

fabaff committed:
Upgrade aiohttp to 3.5.3 (#19957)

* Upgrade aiohttp to 3.5.3

* Upgrade aiohttp to 3.5.4

* Remove test for webhook component from camera.push

* Lint

balloob committed:
Strip login username in backend (#20150)

* Add modern mode to HA auth provider that strips usernames

* Add tests for async_get_or_create_credentials

* Fix test

damarco committed:
Bump pynuki to 1.3.2 (#20173)

syssi committed:
Add Xiaomi Airpurifier Pro V7 support (#20093)

* Add Xiaomi Air Purifier Pro V7 support

* Reorder the model list

* Improve the list of supported attributes/properties

* Fix lint

damarco committed:
Add support for deconz radios to zha component (#20167)

* Add support for deconz radios

* Update check_zigpy_connection()

balloob committed:
Updated frontend to 20190116.0

boralyl committed:
Remove color call to set lights to black. (#20176)

Calling clear all is enough to turn off the light.  Calling the color
command makes the light no longer function until clear all is called
again.  The component calls clear all beforing turning it on which is
why it works through home assistant.  However if you try to control the
light via the hyperion app or through kodi after it has been turned off
via home assistant it will not function until you call clear all again.

balloob committed:
Sensibo to use HA operation modes (#20180)

rohankapoorcom committed:
Embed zoneminder platforms into component (#20182)

balloob committed:
Distribute reconnect (#20181)

balloob committed:
Add command to refresh auth (#20183)

rohankapoorcom committed:
Add a Zoneminder availability sensor (#20184)

* Embed zoneminder platforms into component

* Add a binary sensor for ZoneMinder availability

* Lint

* Add missing docstrings

emontnemery committed:
Add JSON attribute topic to MQTT cover (#20190)

* Add JSON attribute topic to MQTT cover

* Lint

emontnemery committed:
Add JSON attribute topic to MQTT light (#20191)

emontnemery committed:
Add JSON attribute topic to MQTT switch (#20192)

emontnemery committed:
Log exceptions thrown by signal callbacks (#20015)

* Log exceptions thrown by signal callbacks

* Fix unsub

* Simplify traceback print

* Typing

* Add test

* lint

* Review comments

* Rework MQTT test case

* Fix bad merge

* Fix bad merge

ehendrix23 committed:
Set ehendrix23 as owner for harmony platform (#20203)

Put myself (ehendrix23) as code owner for remote.harmony platform

zhujisheng committed:
Add platform image_processing.qrcode (#20215)

* Add platform image_processing.qrcode

* Update qrcode.py

rohankapoorcom committed:
Add verify_ssl to restful_command and switch.rest (#20199) (#20207)

amelchio committed:
Improve Sonos discovery (#20196)

LouisMT committed:
Fix unused friendly name for SolarEdge sensor (#20109)

JoakimLindbom committed:
Fix for issue #19086 (#20225)

emontnemery committed:
Add JSON attribute topic to MQTT vacuum

OttoWinter committed:
Bump aioesphomeapi to 1.4.2 (#20247)

* Bump aioesphomeapi to 1.4.2

* Update requirements_all.txt

gorynychzmey committed:
Use local IP to discover IGD device (#20035)

* Use local_ip from config to discover IGD device

In case of multi-homed server UPNP discovery finds IGD device on some "default" interface. WIth this modification discovery will be performed from 'local_ip'.

* Update device.py

* Changed version of async_upnp_client in requirements

* Used aysnc_upnp_client==0.14.0

* Changed requirement to async_upnp_client==0.14.0.dev0

* Changed requirement to async_upnp_client==0.14.0.dev0

* Changed requirement to async_upnp_client==0.14.0.dev0

* Fixed code style

* Fixed code style

* Changed version of async_upnp_client in requerements

* Changed version of async_upnp_client in requirements

* Regenerated requirements (new async_upnp_client)

* Regenerated requirements (new async_upnp_client)

* Changed requirement to async_upnp_client=0.14.1

* Changed requirement to async_upnp_client=0.14.1

* Updated requirements

* Updated requirements.txt

* Corrected requirements

* Corrected import of DeviceState

* Constants changed according new async_upnp_client

* Upgraded for async_upnp_client==0.14.2

Danielhiversen committed:
Tibber Pulse for homes without subscriptions (#20246)

mwegner committed:
Added Search Configuration to IMAP Sensor (#19749)

* Added Search Configuration to IMAP Sensor

The IMAP sensor currently only counts unread emails in a folder.  By exposing the IMAP search parameter, the sensor can be used to count other results:

- All emails in an inbox
- Emails sent from an address
- Emails matching a subject
- Other advanced searches, especially with vendor-specific extensions.  Gmail in particular supports X-GM-RAW, which lets you use any Gmail search directly ("emails with X label older than 14 days with", etc)

For my use case, I just wanted total emails in a folder, to show an "X/Y" counter for total/unread.  I started work on a one-off script to throw the data in, but figured I'd try to extend Home Assistant more directly, especially since this IMAP sensor correctly handles servers that push data.  This is my first Home Assistant contribution, so apologies in advance if something is out of place!  It's a pretty minimal modification.

* Added Server Response Checking

Looks like no library exception is thrown, so check for response text before parsing out results (previous code just counts spaces, so an error actually returns a state value of 4).

* IMAP Warning -> Error, Count Initializes to None

IMAP search response parsing throws an error instead of a warning.

Email count initializes as None instead 0.

Email count is untouched in case of failure to parse response (i.e. if server is temporarily down or throwing errors, or maybe due to user updating their authentication/login/etc).

Fixed line length on error so it fits under 80 characters.

* Fixed Indent on Logger Error

Sorry about the churn!  Python is pretty far from my daily-use language.  (I did run this one through pep8, at least)

alengwenus committed:
Support for relay ports for LCN light platform (#19632)

* Added relay ports to LCN lights platform

* Exchanged validation for ports with uppercase validator. Makes interfacing with pypck enums much more simple.

* Removed supported_features property as it is correctly inherited from parent

* Removed type annotations.

alengwenus committed:
Add LCN switch platform (#20267)

* Add LCN switch platform

* Added guard clause for discovery_info check and removed unnecessary parathesis

kbickar committed:
Updated sense library to 0.6.0 (#20271)

koreth committed:
Fire events for Lutron RadioRA2 keypad buttons (#20090)

* Add binary sensor for Lutron RadioRA2 keypad buttons

Allow automations to be triggered from RadioRA2 keypads by exposing
each button as a binary sensor.

* Remove binary sensor component; fire events directly instead.

* Address comments from code review

Danielhiversen committed:
Update Tibber library (#20273)

notoriousbdg committed:
Make Netatmo battery_percent icon dynamic (#20275)

scop committed:
Add type hints to helpers.condition (#20266)

amelchio committed:
Remove double logging of automation action (#20264)

alengwenus committed:
Add discovery_info check to LCN light platform (#20280)

* Added discovery_info check to LCN light platform

* Removed whitespaces

balloob committed:
Updated frontend to 20190120.0

koreth committed:
Reduce log noise from Envisalink component (#20282)

amelchio committed:
Fix 'all' entity_id in service call extraction (#20281)

amelchio committed:
Allow 'all' entity_id in service schema (#20278)

rytilahti committed:
Print a message when reconnected after a connection failure, requirement for IQS silver (#20261)

balloob committed:
Add JSON attribute topic to MQTT vacuum (#20242)

emontnemery committed:
Add JSON attribute topic to MQTT lock (#20241)

emontnemery committed:
Add JSON attribute topic to MQTT fan (#20240)

emontnemery committed:
Add JSON attribute topic to MQTT climate (#20239)

emontnemery committed:
Add JSON attribute topic to MQTT alarm (#20238)

pnbruckner committed:
Handle non-string values in JSON renderer (#20233)

Handle the case of async_render_with_possible_json_value's value argument
being something other than a string. This can happen, e.g., when using the
SQL sensor to extract a datetime column such as last_changed and also using
its value_template to convert that datetime to another format. This was
causing a TypeError from json.loads, but async_render_with_possible_json_value
was only catching ValueError's.

balloob committed:
Clean up build artifacts correctly

joncar committed:
Add EverLights light component (#19817)

* EverLights light integration. Supports single color (with color and brightness parameters) or saved pattern (with effect parameter).

* Fix pylint parameter name warning.

* Code review feedback.

* Add tests for the two helper functions of EverLights component.

* Fixes for review feedback.

* Change test style.

* Style fixes for hound.

andrewsayre committed:
Added recursive detection of functools.partial. (#20284)

foxel committed:
Added Xiaomi AirPurifier 2S profile (#20285)

orrpan committed:
Merge branch 'emulated-hue_sleepcycle' of github.com:orrpan/home-assistant into emulated-hue_sleepcycle

549876099 commented:
I can not find my light on SleepCycle app. but it already works.

{
	"1": {
		"modelid": "HASS123",
		"name": "Bedroom Bulb",
		"state": {
			"bri": 255,
			"hue": 4880,
			"on": true,
			"reachable": true,
			"sat": 88
		},
		"swversion": "123",
		"type": "Dimmable light",
		"uniqueid": "light.bedroom_bulb"
	}
}
